-1

**Failed Tests**:   Build 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2d1fd5/13178/summary.html
**COMMIT**: aec34b74407391a0cef825df764203ee3c4a4849
**CMSSW**: CMSSW_11_3_X_2021-03-01-1100/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33029/13178/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_3_X_2021-03-01-1100/src/CondFormats/GEMObjects/src/T_EventSetup_GEMELMap.cc
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_3_X_2021-03-01-1100/src/CondFormats/GEMObjects/src/T_EventSetup_GEMMaskedStrips.cc
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_3_X_2021-03-01-1100/src/CondFormats/GEMObjects/src/T_EventSetup_GEMeMap.cc
>> Compiling  tmp/slc7_amd64_gcc900/src/CondFormats/GEMObjects/src/CondFormatsGEMObjects/a/Serialization.cc
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_3_X_2021-03-01-1100/src/CondFormats/GEMObjects/src/GEMELMap.cc: In member function 'void GEMELMap::convertDummy(GEMROmap&)':
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_3_X_2021-03-01-1100/src/CondFormats/GEMObjects/src/GEMELMap.cc:66:48: error: 'maxRollId' is not a member of 'GEMDetId'
   66 |           for (int roll = 1; roll <= GEMDetId::maxRollId; ++roll) {
      |                                                ^~~~~~~~~
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_3_X_2021-03-01-1100/src/CondFormats/GEMObjects/src/GEMeMap.cc: In member function 'void GEMeMap::convertDummy(GEMROMapping&)':
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_3_X_2021-03-01-1100/src/CondFormats/GEMObjects/src/GEMeMap.cc:109:50: error: 'maxRollId' is not a member of 'GEMDetId'
  109 |             for (int roll = 1; roll <= GEMDetId::maxRollId; ++roll) {
</pre>

--- a/DataFormats/MuonDetId/interface/GEMDetId.h
+++ b/DataFormats/MuonDetId/interface/GEMDetId.h
@@ -32,6 +32,8 @@ public:
   static constexpr int32_t maxLayerId = 2;  // GE1/GE2 has 2 layers
   static constexpr int32_t minEtaPartitionId = 0;
   static constexpr int32_t maxEtaPartitionId = 16;
+  static constexpr int32_t minRollId = minEtaPartitionId;
+  static constexpr int32_t maxRollId = maxEtaPartitionId;
 
 private:
   static constexpr uint32_t RegionNumBits = 2;

